node/repo/blockstore_opts.go

```diff
@@ -21,9 +21,8 @@ func BadgerBlockstoreOptions(domain BlockstoreDomain, path string, readonly bool
 	opts.DetectConflicts = false
 
 	// This is to optimize the database on close so it can be opened
-	// read-only and efficiently queried. We don't do that and hanging on
-	// stop isn't nice.
-	opts.CompactL0OnClose = false
+	// read-only and efficiently queried.
+	opts.CompactL0OnClose = true
```

Comment:
Can this be problematic/lead to data loss if user triggers shutdown, and after a few seconds seeing that the process isn't stopping, kills the process during compaction?

Reply:
This is how Lotus used to behave before the native badger blockstore: https://github.com/filecoin-project/lotus/blob/d4cdc6d3340b8496c9f98e2d0daed8d1bd9b271e/node/repo/fsrepo_ds.go#L34-L35

The default value of this flag is `true`.

Flipping it to `false` was actually a regression, and it was caught by the Sentinel team because their lens component opens the store in read-only mode. Since L0 (unsorted level) hadn't been compacted to L1 (sorted level), badger refused to open it.

If we didn't notice corruption issues before, we shouldn't notice them now. However, I won't put my hand in the fire for badger...